google-cloud-logging/src/main/java/com/google/cloud/logging/LoggingHandler.java

```diff
@@ -457,28 +500,31 @@ void write(LogEntry entry, WriteOption... options) {
           reportError(null, ex, ErrorManager.FLUSH_FAILURE);
         }
         break;
+
       case ASYNC:
       default:
-        ApiFutures.addCallback(getLogging().writeAsync(entryList, options), new ApiFutureCallback<Void>() {
-          @Override
-          public void onSuccess(Void v) {}
-
-          @Override
-          public void onFailure(Throwable t) {
-            if (t instanceof Exception) {
-              reportError(null, (Exception) t, ErrorManager.FLUSH_FAILURE);
-            } else {
-              reportError(null, new Exception(t), ErrorManager.FLUSH_FAILURE);
-            }
-          }
-        });
+        flushMonitor.enterWhenUninterruptibly(noFlushers);
```

Comment:
This is correct. Otherwise, new writes will keep incrementing the `numPendingWrites` and flushes might never complete. There are ways around it, but I think this is fine for now, since flushes should happen relatively rarely. (Or so it is hoped anyway.)